### ISSUE TITLE
False positive for no-unused-vars, when function is annotated as `exported`

### ISSUE THREAD

**Issue by mcandre** [eslint/eslint#11813]
**Tell us about your environment**

* **ESLint Version:**

v5.16.0

* **Node Version:**

v10.15.3

* **npm Version:**

6.9.0

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

```js
{
    "env": {
        "node": false,
        "browser": true,
        "mocha": true,
        "jquery": true
    },
    "rules": {
        "global-strict": true,
        "no-mixed-requires": 2,
        "no-unused-vars": [2, { "args": "none" }],
        "quotes": [2, "single"]
    }
}
```

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

ajax-errors.js:

```js
/* exported: handleError */
function handleError(jqXHR, textStatus) {
    console.log('Error: ' + textStatus);
}
```

ajax.handlebars:

```js
$.ajax({
    url: '/ajax/activity/list',
    method: 'POST',
    'dataType': 'json',
    'contentType': 'application/json',
    'data': JSON.stringify({
        budget_id: {{id}},
        sort: 2,
        limit: 4
    }),
    error: handleError,
    success: function(data) {
        var rows = data;

        var t = $('#activities_{{id}} table');

        for (var i = 0; i < rows.length && i < 4; i++) {
            var row = rows[i];
            t.append('<tr><td>' + row['t'] + '</td><td>' + row['message'] + '</td></tr>');
        }
    }
});
```

```bash
$ "$(npm bin)/eslint" .

public/ajax-errors.js
  7:10  error  'handleError' is defined but never used  no-unused-vars

✖ 1 problem (1 error, 0 warnings)
```

**What did you expect to happen?**

I expected my code to pass eslint with zero findings.

**What actually happened? Please include the actual, raw output from ESLint.**

eslint complained that handleError was unused. Even though I carefully marked it with an exported annotation.

In fact, I do use this function when I pass it to the `error` parameter of a jQuery `ajax()` call. True, this call is obfuscated by the fact that the use is inside of a Handlebars template. But the fact that I annotate this binding as `exported` should be sufficient to inform eslint not to report this as "unused".

**Comment by mysticatea** [comment]
Looks like the colon is extra:

```diff
- /* exported: handleError */
+ /* exported handleError */
```

**Comment by mcandre** [comment]
:P

Correct.
